auraed/src/runtime/cell_service/cells/cell.rs

```diff
@@ -0,0 +1,183 @@
+/* -------------------------------------------------------------------------- *\
+ *             Apache 2.0 License Copyright © 2022 The Aurae Authors          *
+ *                                                                            *
+ *                +--------------------------------------------+              *
+ *                |   █████╗ ██╗   ██╗██████╗  █████╗ ███████╗ |              *
+ *                |  ██╔══██╗██║   ██║██╔══██╗██╔══██╗██╔════╝ |              *
+ *                |  ███████║██║   ██║██████╔╝███████║█████╗   |              *
+ *                |  ██╔══██║██║   ██║██╔══██╗██╔══██║██╔══╝   |              *
+ *                |  ██║  ██║╚██████╔╝██║  ██║██║  ██║███████╗ |              *
+ *                |  ╚═╝  ╚═╝ ╚═════╝ ╚═╝  ╚═╝╚═╝  ╚═╝╚══════╝ |              *
+ *                +--------------------------------------------+              *
+ *                                                                            *
+ *                         Distributed Systems Runtime                        *
+ *                                                                            *
+ * -------------------------------------------------------------------------- *
+ *                                                                            *
+ *   Licensed under the Apache License, Version 2.0 (the "License");          *
+ *   you may not use this file except in compliance with the License.         *
+ *   You may obtain a copy of the License at                                  *
+ *                                                                            *
+ *       http://www.apache.org/licenses/LICENSE-2.0                           *
+ *                                                                            *
+ *   Unless required by applicable law or agreed to in writing, software      *
+ *   distributed under the License is distributed on an "AS IS" BASIS,        *
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. *
+ *   See the License for the specific language governing permissions and      *
+ *   limitations under the License.                                           *
+ *                                                                            *
+\* -------------------------------------------------------------------------- */
+
+use super::{CellName, CellSpec, CellsError, Cgroup, Result};
+use crate::runtime::cell_service::executables::auraed::NestedAuraed;
+use aurae_client::AuraeConfig;
+use tracing::info;
+
+// We should not be able to change a cell after it has been created.
+// You must free the cell and create a new one if you want to change anything about the cell.
+// In order to facilitate that immutability:
+// NEVER MAKE THE FIELDS PUB (OF ANY KIND)
+#[derive(Debug)]
+pub struct Cell {
+    name: CellName,
+    spec: CellSpec,
+    state: CellState,
+}
+
+// TODO: look into clippy warning
```

Comment:
tl;dr: it's due to an imbalance between the three variants where one is _much larger_ than the other, creating a potential issue with the memory layout.

i think it's ok to ignore this one as i don't see a better way of doing it (boxing the fields would be plausible but clunky).